src/trio/_sync.py

```diff
@@ -112,6 +113,17 @@ def statistics(self) -> EventStatistics:
         """
         return EventStatistics(tasks_waiting=len(self._tasks))
 
+    def __bool__(self) -> True:
+        """Return True and raise warning."""
+        warn_deprecated(
+            self.__bool__,
+            "0.30.1",
+            issue=3238,
+            instead=self.is_set,
+            use_triodeprecationwarning=True,
```

Comment:
This should be false

Reply:
Added test and not using legacy deprecation warning in 68a4fad